### PR TITLE
Enrollment CRUD and admin pages

### LINKED ISSUE #82: Enrollment CRUD operations/admin tools
We should implement CRUD operations for Enrollments, along with appropriate admin tools. It will allow us to monitor each participants progress of earning points, detect eventual anomalies, and we could also edit an enrollment to easily swap it to another account (in case someone forgets their password or something and creates a new account).

### PR DESCRIPTION
**Pull request by pedrosilva17**
Closes #82
Implements CRUD for enrollments and admin pages to manage them.

### CODE REVIEW

File: app/Http/Controllers/EnrollmentCRUDController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class EnrollmentCRUDController extends CRUDController
+{
+    protected string $model = Enrollment::class;
+
+    protected string $view = 'Enrollment';
+
+    protected array $rules = [
+        'student_id' => 'required|exists:students,id',
+        'edition_id' => 'required|exists:editions,id',
+        'points' => 'required|integer',
```
Comment (ttoino) [inline app/Http/Controllers/EnrollmentCRUDController.php]: A user's points are calculated automatically, they shouldn't be editable.

**Review by ttoino — CHANGES_REQUESTED** [review]

**Review by toni-santos — DISMISSED** [review]

File: resources/js/Pages/CRUD/Enrollment/Create.vue [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <select v-model="form.edition_id" required class="self-stretch">
+                <option value="" disabled selected hidden>Edição</option>
+                <option
+                    v-for="edition in $props.with.editions"
+                    :key="edition.id"
+                    :value="edition.id"
+                >
+                    {{ edition.name }}
+                </option>
+            </select>
```
Comment (Naapperas) [inline resources/js/Pages/CRUD/Enrollment/Create.vue]: Use the new TextInput type `select`: there are already examples in place using that component.

File: resources/js/Pages/CRUD/Enrollment/Edit.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                <option value="" disabled selected hidden>Edição</option>
+                <option
+                    v-for="edition in $props.with.editions"
+                    :key="edition.id"
+                    :value="edition.id"
+                >
+                    {{ edition.name }}
+                </option>
+            </select>
+
```
Comment (Naapperas) [inline resources/js/Pages/CRUD/Enrollment/Edit.vue]: Same as above.

File: resources/js/Pages/CRUD/Enrollment/Index.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+</script>
+
+<template>
+    <CRUDLayout title="Enrollment" :items="items" name="enrollments">
+        <template #heading>Inscrições</template>
+
+        <template #header>
+            <HeaderRow>
+                <Header sort-by="participant_id">Nome do Estudante</Header>
+                <Header sort-by="edition_id">Nome da Edição</Header>
```
Comment (Naapperas) [inline resources/js/Pages/CRUD/Enrollment/Index.vue]: Change this to a `filter-by`.

**Review by Naapperas — CHANGES_REQUESTED** [review]

**Review by Naapperas — APPROVED** [review]

**Review by ttoino — APPROVED** [review]

**Pull request merged**